iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -434,20 +355,29 @@ bool ProcessManager::addProcess(const ProcessName_t& name,
     return true;
 }
 
+bool ProcessManager::unregisterProcess(const ProcessName_t& name) noexcept
+{
+    std::lock_guard<std::mutex> g(m_mutex);
+    if (!removeProcess(name))
+    {
+        LogError() << "Application " << name << " could not be unregistered!";
+        return false;
+    }
+    return true;
+}
+
 bool ProcessManager::removeProcess(const ProcessName_t& name) noexcept
 {
-    std::lock_guard<std::mutex> lockGuard(m_mutex);
     // we need to search for the process (currently linear search)
-
     auto it = m_processList.begin();
     while (it != m_processList.end())
     {
         auto otherName = it->getName();
         if (name == otherName)
         {
-            if (removeProcess(lockGuard, it))
```

Comment:
please don't remove the safety net with the lock guard

Reply:
Locking is now done via `smart_lock` around the `ProcessManager` object.